bowtie/_report.py

```diff
@@ -288,6 +294,57 @@ def empty(
             did_fail_fast=False,
         )
 
+    @classmethod
+    def combine_versioned_reports_for(
+        cls,
+        versioned_reports: Iterable[Report],
+        dialect: Dialect,
+    ) -> Report | None:
+        versioned_reports = [
+            versioned_report
+            for versioned_report in versioned_reports
+            if versioned_report.metadata.dialect == dialect
+        ]
+
+        if versioned_reports:
+
+            cases: HashTrieMap[Seq, TestCase] = HashTrieMap()
+            results: HashTrieMap[
+                ConnectableId,
+                HashTrieMap[Seq, SeqResult],
+            ] = HashTrieMap()
+            implementations: dict[ConnectableId, ImplementationInfo] = {}
+
+            for versioned_report in versioned_reports:
+                for seq, case in versioned_report._cases.items():
+                    existing_case = cases.get(seq)
+                    if not existing_case:
+                        cases = cases.insert(seq, case)
+                    elif case != existing_case:
+                        raise CaseMismatchInReportsForSameDialect()
+
+                version_id, version_results = next(
+                    iter(versioned_report._results.items()),
+                )
+                results = results.insert(version_id, version_results)
+
+                version_id, version_info = next(
+                    iter(versioned_report.metadata.implementations.items()),
+                )
+                implementations[version_id] = version_info
+
+            return cls(
+                cases=cases,
+                results=results,
+                metadata=RunMetadata(
+                    implementations=implementations,
+                    dialect=dialect,
+                ),
+                did_fail_fast=False,
+            )
+        else:
```

Comment:
Can you invert the order of the if blocks (to `if not versioned_reports`) here which will let you dedent the larger block